contrib/dockerfile_build.bzl

```diff
@@ -0,0 +1,65 @@
+# Copyright 2016 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository rule to build image with `docker build`."""
+
+def _impl(rctx):
+  tag = rctx.attr.tag
+  dockerfile = rctx.path(rctx.attr.dockerfile)
+  result = rctx.execute([
```

Comment:
This function is not documented in here https://docs.bazel.build/versions/master/skylark/lib/ctx.html.
I am assuming this executes a command as a subprocess?

Reply:
Drive-by comment: object is a repository_ctx object, not "regular" context object.

https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#execute